### ISSUE TITLE
printf: %a not supported (but partially implemented)

### ISSUE THREAD

**Issue by jfinkels** [uutils/coreutils#2776]
```
$ printf "%a\n" 255
0xf.fp+4
$ ./target/debug/coreutils printf "%a\n" 255
printf: %a: invalid conversion specification
```

Here is the description of the floating-point conversions: https://www.gnu.org/software/libc/manual/html_node/Floating_002dPoint-Conversions.html

This seems to be called `CninetyNineHexFloatf` in the code: https://github.com/uutils/coreutils/blob/ca705e5ad2bc6b0b0c5dde0d0c1df38bcb0f2148/src/uu/printf/src/tokenize/num_format/formatters/cninetyninehexfloatf.rs#L11 but it seems to be only partially implemented and not exposed to the user: https://github.com/uutils/coreutils/blob/ca705e5ad2bc6b0b0c5dde0d0c1df38bcb0f2148/src/uu/printf/src/tokenize/sub.rs#L163

**Comment by tertsdiepraam** [comment]
There's been progress on this in https://github.com/uutils/coreutils/pull/5128, but it's not quite correct yet. The current output of you example is
```
0x1.fe00000000000p+7
```

**Comment by jfinkels** [comment]
Some observations from experimenting with `printf %a` for various integer values.

* It looks like any rendered with `%a` must have 8, 9, a, b, c, d, e, or f on the left of the decimal point (hexadecimal point?). The digits 0 through 7 are unused. For example, `printf "%a 31` is `0xf.8p+1` and `printf %a 32` is ` 0x8p+2`.
* If no fractional digits are needed, there is no decimal point. For example, `printf %a 18` is `0x9p+1`.
* The number of digits after the decimal point is the minimum possible, increasing as needed. For example, `printf %a 257` is `0x8.08p+5` but `printf %a 258` is `0x8.1p+5`.

**Comment by jfinkels** [comment]
It looks like the function to update now is `format_float_hexadecimal()` in `src/uucore/src/lib/features/format/num_format.rs`. No idea what the right algorithm is though!

**Comment by drinkcat** [comment]
Oh, sorry, I didn't notice this issue. I dug into this in #7364 as well. Behaviour is platform-dependent, and we can't get the same precision as x86(-64) by just using Rust's `f64`.

**Comment by jtracey** [comment]
I think this was fixed in #7631?

**Comment by drinkcat** [comment]
Yeah, to the extend this can be "fixed". See #7641 for the documentation of the current behaviour.

**Comment by RenjiSann** [comment]
I will close this issue, since the feature was implemented (albeit the conversion isn't a 1:1 match with GNU implementation)
